source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/ParserRegistrationTest.cpp

```diff
@@ -0,0 +1,105 @@
+#include "stdafx.h"
+#include "CppUnitTest.h"
+#include "TextBlock.h"
+#include <time.h>
+#include <Windows.h>
+#include <StrSafe.h>
+#include "SharedAdaptiveCard.h"
+#include "BaseCardElement.h"
+#include "ActionParserRegistration.h"
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+using namespace AdaptiveCards;
+using namespace std::string_literals;
+
+namespace AdaptiveCardsSharedModelUnitTest
+{
+	TEST_CLASS(ParserRegistrationTest)
+	{
+	public:
+		TEST_METHOD(ParserRegistrationTests)
+		{
+			// Define custom type. This implements both element and action for convenience
+			class TestCustomElement : public BaseCardElement, public BaseActionElement
+			{
+			public:
+				TestCustomElement(
+					const Json::Value& value) :
+					BaseCardElement(AdaptiveCards::CardElementType::Custom),
+					BaseActionElement(AdaptiveCards::ActionType::Custom)
+				{
+					m_customImage = value.get("customImageProperty", Json::Value()).asString();
+				}
+
+				virtual void GetResourceInformation(std::vector<RemoteResourceInformation>& resourceUris) override
+				{
+					RemoteResourceInformation resourceInfo;
+					resourceInfo.url = m_customImage;
+					resourceInfo.resourceType = CardElementType::Image;
+					resourceUris.push_back(resourceInfo);
+				}
+
+			private:
+				std::string m_customImage;
+			};
+
+			// Define custom element parser
+			class TestCustomParser : public BaseCardElementParser
+			{
+			public:
+				virtual std::shared_ptr<BaseCardElement> Deserialize(
+					std::shared_ptr<AdaptiveCards::ElementParserRegistration> elementParserRegistration,
+					std::shared_ptr<AdaptiveCards::ActionParserRegistration> actionParserRegistration,
+					std::vector<std::shared_ptr<AdaptiveCards::AdaptiveCardParseWarning>>& warnings,
+					const Json::Value& value) override
+				{
+					return std::make_shared<TestCustomElement>(value);
+				}
+			};
+
+			// Define custom action parser
+			class TestCustomActionParser : public ActionElementParser
+			{
+			public:
+				virtual std::shared_ptr<BaseActionElement> Deserialize(
+					std::shared_ptr<AdaptiveCards::ElementParserRegistration> elementParserRegistration,
+					std::shared_ptr<AdaptiveCards::ActionParserRegistration> actionParserRegistration,
+					std::vector<std::shared_ptr<AdaptiveCards::AdaptiveCardParseWarning>>& warnings,
+					const Json::Value& value) override
+				{
+					return std::make_shared<TestCustomElement>(value);
+				}
+			};
+
+			ActionParserRegistration parser;
```

Comment:
>ActionParserRegistration [](start = 3, length = 24)

Shall we test element parser registration as well?

Reply:
I was going to get to it in a future checkin, but I suppose I can just add it in now 👍